controllers/toolchaincluster/toolchaincluster_controller_test.go

```diff
@@ -103,26 +104,72 @@ func TestClusterControllerChecks(t *testing.T) {
 		// then
 		require.Equal(t, err, nil)
 		require.Equal(t, reconcile.Result{RequeueAfter: requeAfter}, recresult)
-		assertClusterStatus(t, cl, "stable", healthy())
+		assertClusterStatus(t, cl, "stable", clusterReadyCondition())
 	})
 
 	t.Run("toolchain cluster cache not found", func(t *testing.T) {
 		// given
-		stable, _ := newToolchainCluster("stable", tcNs, "http://cluster.com", toolchainv1alpha1.ToolchainClusterStatus{})
+		unstable, _ := newToolchainCluster("unstable", tcNs, "http://cluster.com", toolchainv1alpha1.ToolchainClusterStatus{})
 
-		cl := test.NewFakeClient(t, stable)
+		cl := test.NewFakeClient(t, unstable)
+		unstable.Status.Conditions = condition.AddOrUpdateStatusConditionsWithLastUpdatedTimestamp(unstable.Status.Conditions, clusterNotReadyCondition())
+		controller, req := prepareReconcile(unstable, cl, requeAfter)
 
+		// when
+		_, err := controller.Reconcile(context.TODO(), req)
+
+		// then
+		require.EqualError(t, err, "cluster unstable not found in cache")
+		actualtoolchaincluster := &toolchainv1alpha1.ToolchainCluster{}
+		err = cl.Client.Get(context.TODO(), types.NamespacedName{Name: "unstable", Namespace: tcNs}, actualtoolchaincluster)
+		require.NoError(t, err)
+		assertClusterStatus(t, cl, "unstable", clusterNotReadyCondition())
+	})
+
+	t.Run("error while updating a toolchain cluster status on cache not found", func(t *testing.T) {
+		// given
+		stable, _ := newToolchainCluster("stable", tcNs, "http://cluster.com", withStatus(clusterReadyCondition()))
+
+		cl := test.NewFakeClient(t, stable)
+		cl.MockStatusUpdate = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.UpdateOption) error {
+			return fmt.Errorf("mock error")
+		}
 		controller, req := prepareReconcile(stable, cl, requeAfter)
 
 		// when
-		_, err := controller.Reconcile(context.TODO(), req)
+		recresult, err := controller.Reconcile(context.TODO(), req)
```

Comment:
let's use a camelCase for variable names
```suggestion
		recResult, err := controller.Reconcile(context.TODO(), req)
```